File: DevExpress-Demo/VB/RibbonDemo.Wpf/Modules/RibbonMerging/ViewModels/TextPanelViewModel.vb
Imports DevExpress.Mvvm.POCO
Imports System.Windows
Imports DevExpress.Mvvm.DataAnnotations

Namespace RibbonDemo

    <POCOViewModel>
    Public Class TextPanelViewModel
        Inherits PanelViewModel

        Protected Sub New()
        End Sub

        Public Shared Function Create(ByVal caption As String, ByVal location As Point, ByVal parentViewModel As RibbonMergingViewModel) As TextPanelViewModel
            Dim instance As TextPanelViewModel = ViewModelSource.Create(Function() New TextPanelViewModel())
            instance.ParentViewModel = parentViewModel
            instance.Caption = caption
            instance.Location = location
            Return instance
        End Function
    End Class
End Namespace

File: DevExpress-Demo/CS/RibbonDemo.Wpf/Modules/RibbonMerging/ViewModels/TextPanelViewModel.cs
using DevExpress.Mvvm.POCO;
using System.Windows;
using DevExpress.Mvvm.DataAnnotations;

namespace RibbonDemo {
    [POCOViewModel]
    public class TextPanelViewModel : PanelViewModel {
        protected TextPanelViewModel() { }
        public static TextPanelViewModel Create(string caption, Point location, RibbonMergingViewModel parentViewModel) {
            TextPanelViewModel instance = ViewModelSource.Create(() => new TextPanelViewModel());
            instance.ParentViewModel = parentViewModel;
            instance.Caption = caption;
            instance.Location = location;
            return instance;
        }
    }
}
